s3tests_boto3/functional/test_s3select.py

```diff
@@ -1240,51 +1275,45 @@ def test_progress_expressions():
+    res_s3select_1 = remove_xml_tags_from_result(  run_s3select_output(bucket_name,csv_obj_name,"select _1, _2 from s3object where nullif(_1,_2) is null ;", "ALWAYS")  ).replace("\n",",")
 
-    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,"select _1, _2 from s3object where _1 = _2 ;")  ).replace("\n","")
+    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,"select _1, _2 from s3object where _1 = _2 ;")  ).replace("\n",",")
 
     res_s3select_list = res_s3select.split(',')
-
-    res_s3select_list.pop()
-
-    res_s3select_final = (','.join('"' + item + '"' for item in res_s3select_list))
-
-    res_s3select_final += ','
+    res_s3select_final = (','.join('"' + item + '"' for item in res_s3select_list)).replace('""','') # remove empty result(first,last)
 
     s3select_assert_result( res_s3select_1, res_s3select_final)
 
     res_s3select_in = remove_xml_tags_from_result(  run_s3select_output(bucket_name,csv_obj_name,'select int(_1) from s3object where (int(_1) in(int(_2)));', "ASNEEDED", '$', '#')).replace("\n","")
 
-    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select int(_1) from s3object where int(_1) = int(_2);')).replace("\n","")
-    
-    res_s3select_list = res_s3select.split(',')
-
-    res_s3select_list.pop()
-
-    res_s3select_final = ('#'.join(item + '$' for item in res_s3select_list))
-
-    res_s3select_final += '#'
+    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select int(_1) from s3object where int(_1) = int(_2);')).replace("\n","#")
+    res_s3select = res_s3select[1:len(res_s3select)] # remove first redundant
+    res_s3select_final = res_s3select[0:len(res_s3select)-1] # remove last redundant
 
     s3select_assert_result( res_s3select_in, res_s3select_final )
 
     res_s3select_quot = remove_xml_tags_from_result(  run_s3select_output(bucket_name,csv_obj_name,'select int(_1) from s3object where (int(_1) in(int(_2)));', "ALWAYS", '$', '#')).replace("\n","")
 
-    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select int(_1) from s3object where int(_1) = int(_2);')).replace("\n","")
+    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select int(_1) from s3object where int(_1) = int(_2);')).replace("\n","#")
+    res_s3select = res_s3select[1:len(res_s3select)] # remove first redundant
+    res_s3select = res_s3select[0:len(res_s3select)-1] # remove last redundant
+
+    res_s3select_list = res_s3select.split('#')
+    res_s3select_final = ('#'.join('"' + item + '"' for item in res_s3select_list)).replace('""','')
     
-    res_s3select_list = res_s3select.split(',')
+    s3select_assert_result( res_s3select_quot, res_s3select_final )
 
-    res_s3select_list.pop()
+@attr('s3select')
+def test_parqueet():
```

Comment:
So all this test is doing is generating the parquet file but not uploading it?

Reply:
yes,
I checked that the installation for pyarrow and pandas works.
later, the test will upload parquet to RGW and will run queries on it.